UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -802,6 +802,52 @@ protected virtual void BuildAnimationSamplers(GLTFAnimation animation, int anima
 			GLTFHelpers.BuildAnimationSamplers(ref samplersByType);
 		}
 
+		public delegate float[] ValuesConvertion(NumericArray data, int frame);
+        protected void SetAnimationCurve(
+            AnimationClip clip,
+            string relativePath,
+            string[] propertyNames,
+            NumericArray input,
+            NumericArray output,
+            InterpolationType mode,
+            Type curveType,
+			ValuesConvertion getConvertedValues)
+        {
+
+			var channelCount = propertyNames.Length;
+			var frameCount = input.AsFloats.Length;
+			
+			// copy all the key frame data to cache
+			List<Keyframe>[] keyframes = new List<Keyframe>[channelCount];						
```

Comment:
nit- You can just make this 2D array instead of a List array

Reply:
Fixed it in my 2nd commit.